grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationAtom.java

```diff
@@ -489,12 +517,27 @@ public Atom inferTypes() {
     }
 
     /**
-     * @return set constituting the role player var names that do not have a specified role type
+     * @return rolePlayer-roleVariable pairs that are not specific (either don't have a role pattern or have a meta role)
      */
-    private Set<Var> getNonSpecificRolePlayers() {
-        Set<Var> unmappedVars = getRolePlayers();
-        unmappedVars.removeAll(getSpecificRolePlayers());
-        return unmappedVars;
+    private Set<Pair<Var, Var>> getNonSpecificRelationPlayers() {
+        return getRelationPlayers().stream()
+                .filter(rp -> rp.getRole().isPresent())
+                .filter(rp -> {
+                    VarPatternAdmin rolePattern = rp.getRole().orElse(null);
+                    if (rolePattern == null) return true;
+
+                    Label roleLabel = rolePattern.getTypeLabel().orElse(null);
+                    return roleLabel == null || Schema.MetaSchema.isMetaLabel(roleLabel);
```

Comment:
This null check `roleLabel == null` is not needed. `Schema.MetaSchema.isMetaLabel(roleLabel)` can take in a null and fail as expected.

Reply:
if `roleLabel` is null I need it to return `true`. If I pass null to `isMetaLabel` it returns `false`.